File: 品味Java的21種設計模式/dp_cpp/command/example5/ChopCommand.cpp
#include "ChopCommand.h"

namespace cn
{
	namespace javass
	{
		namespace dp
		{
			namespace command
			{
				namespace example5
				{

					void ChopCommand::setCookApi(CookApi *cookApi)
					{
						this->cookApi = cookApi;
					}

					void ChopCommand::execute()
					{
						this->cookApi->cook("бл╤╧ее╧гЛр");
					}
				}
			}
		}
	}
}

File: 品味Java的21種設計模式/dp_java/command/example5/ChopCommand.java
package cn.javass.dp.command.example5;
/**
 * 命令对象，绿豆排骨煲
 */
public class ChopCommand implements Command{
	/**
	 * 持有具体做菜的厨师的对象
	 */
	private CookApi cookApi = null;
	/**
	 * 设置具体做菜的厨师的对象
	 * @param cookApi 具体做菜的厨师的对象
	 */
	public void setCookApi(CookApi cookApi) {
		this.cookApi = cookApi;
	}
	
	public void execute() {
		this.cookApi.cook("绿豆排骨煲");
	}
}
